GetIntoTeachingApi/Controllers/SchoolsExperience/CandidatesController.cs

```diff
@@ -76,7 +76,7 @@ public class CandidatesController : ControllerBase
                 // Usually, it is best practice to allow the CRM to generate sequential GUIDs which provide better
                 // SQL performance. However, in this scenario we have agreed it is beneficial to provide the GUID up-front
                 // because the School Experience app needs the Candidate ID immediately.
-                candidate.Id = Guid.NewGuid();
+                ((ICanGenerateUpfrontId)candidate).GenerateId();
```

Comment:
Do we need the casts to the interface here and on `MockModel` any more since `BaseModel` implements it?